FrontEnd/src/pages/Products/Products.tsx

```diff
@@ -21,23 +36,56 @@ function Products() {
       genders.push(genderName)
     }
   }
+
   let url = gender
+  const pagingQueries = `limit=9&page=${pageNumber}`
   if (gender === "products") {
     url = ""
     if (genders.length) {
       url += `?${genders.map((gender) => `gender=${gender}`).join("&")}`
     }
   }
-  const { data } = useQuery({
-    queryFn: () => {
-      return ProductAPI.getProducts(url)
-    },
-    queryKey: [gender, url],
-    suspense: true,
+  url += url.includes("?") ? `&${pagingQueries}` : `?${pagingQueries}`
+
+  const { data, isLoading } = useQuery({
+    queryFn: () => ProductAPI.getProducts(url),
+    queryKey: [gender, pageNumber],
     staleTime: Infinity,
-  }) as { data: ProductType[] }
+  })
+
+  useEffect(() => {
+    if (data) {
+      setScrollingData((prevData) => ({
+        ...data,
+        response: [...prevData.response, ...data.response],
+      }))
+
+      if (data.next) {
+        setHasMore(true)
+      } else {
+        setHasMore(false)
+      }
+    }
+  }, [data])
+
+  const lastProductElementRef = useCallback(
+    (node: HTMLLIElement | null) => {
+      if (isLoading) return
+
+      if (observer.current) observer.current.disconnect()
+
+      observer.current = new IntersectionObserver((entries) => {
```

Comment:
You can specify the root margin of when the observer callback fn will be called, in your case the callback fn will be called  right after you intersect the beggining of item you observe, only after that you will fetch new portion of data. That`s not good UX, due to time that needed to fetch new data. So as an option you can specify rootMargin to avoid that behaviour.